plugin/pathvirt/pathvirt.cpp

```diff
@@ -0,0 +1,243 @@
+ */
+
+/*
+ * clfind - returns first index in colonlist which is a prefix for path
+ *          modifies the @listPtr to point to the element in colonlist
+ */
+static int
+clfind(const char *colonlist,  // IN
+       const char *path,       // IN
+       char **listPtr)         // OUT
+{
+    int index = 0;
+    char *element = const_cast<char *>(colonlist);
+    char *colon = NULL;
+
+    /* while there is a colon present, loop */
+    while (colon = strchr(element, ':')) {
+        /* check if element is a prefix of path. here, colon - element is
+           an easy way to calculate the length of the element in the list
+           to use as the size parameter to strncmp */
+        if (strncmp(path, element, colon - element) == 0) {
+            *listPtr = element;
+            return index;
+        }
+
+        /* move element to point to next element */
+        element = colon + 1;
+
+        index++;
+    }
+
+    /* process the last element in the list */
+    if (strncmp(path, element, strlen(element)) == 0) {
+        *listPtr = element;
+        return index;
+    }
+
+    /* not found */
+    return -1;
+}
+
+/*
+ * clget - returns pointer to element in colonlist at index i
+ *         and NULL if not found
+ */
+static char*
+clget(const char *colonlist, unsigned int i)
+{
+    int curr_ind = 0;
+    char *element = const_cast<char *>(colonlist);
+    char *colon = NULL;
+
+    /* iterate through elements until last one */
+    while (colon = strchr(element, ':')) {
+        /* if we are at the request index, return pointer to start of element */
+        if (curr_ind == i)
+            return element;
+
+        /* otherwise, advance pointer to next element and bump current index */
+        element = colon + 1;
+        curr_ind++;
+    }
+
+    /* last element */
+    if (curr_ind == i)
+        return element;
+
+    /* not found */
+    return NULL;
+}
+
+/*
+ * clgetsize_ptr - returns size of an element pointed to by @element in the
+ *                 list
+ */
+static size_t
+clgetsize_ptr(const char *colonlist, const char *element)
+{
+    /* either calculate the element's length, or call
+     * strlen if element was last one */
+    const char *colon = strchr(element, ':');
+    return colon ? colon - element : strlen(element);
+}
+
+/*
+ * clgetsize - returns size of an element at index i in colonlist
+ *             and -1 if not found
+ */
+static ssize_t
+clgetsize_ind(const char *colonlist, const unsigned int i)
+{
+    /* get pointer to element at index i */
+    char *element = clget(colonlist, i);
+    if (element) {
+        /* now that we have a pointer, we can use clgetsize_ptr */
+        return clgetsize_ptr(colonlist, element);
+    }
+
+    /* not found */
+    return -1;
+}
+
+/*
+ * dynamic_path_swap - given old path, return new path or NULL
+ *
+ * Returns NULL if no swap is to be done and the original path value should
+ * be used. Returns a malloc'd pointer to the new string if a swap should
```

Comment:
@rohgarg and @mossberg, thanks for removing the malloc from the code.  Could you now remove the malloc reference from the comment above?  :-)